assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js

```diff
@@ -26,19 +26,35 @@ import { __ } from '@wordpress/i18n';
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
 import { STORE_NAME } from '../../datastore/constants';
 import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
 import Switch from '../../../../components/Switch';
 import { trackEvent } from '../../../../util';
+import { useFeature } from '../../../../hooks/useFeature';
 const { useSelect, useDispatch } = Data;
 
 export default function UseUAandGA4SnippetSwitches() {
+	const isGA4Enabled = useFeature( 'ga4setup' );
+
 	const useUASnippet = useSelect( ( select ) => select( STORE_NAME ).getUseSnippet() );
 	const canUseUASnippet = useSelect( ( select ) => select( STORE_NAME ).getCanUseSnippet() );
 
-	const ga4ExistingTag = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getExistingTag() );
-	const ga4MeasurementID = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getMeasurementID() );
-	const useGA4Snippet = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getUseSnippet() );
+	const {
+		ga4ExistingTag,
+		ga4MeasurementID,
+		useGA4Snippet,
+	} = useSelect( ( select ) => {
+		if ( ! isGA4Enabled || select( CORE_MODULES ).isModuleConnected( 'analytics-4' ) === false ) {
```

Comment:
Related to what you did above, I think it would technically be more appropriate to do the following check here:
```suggestion
		if ( ! isGA4Enabled || select( CORE_MODULES ).isModuleConnected( 'analytics-4' ) !== select( CORE_MODULES ).isModuleConnected( 'analytics' ) ) {
```

Not strictly necessary right now, but e.g. if we wanted to show these toggles in the setup flow too it would become a problem.

Reply:
Updated.